keras_cv/layers/preprocessing/random_cutout.py

```diff
@@ -0,0 +1,248 @@
+from tensorflow.python.keras.utils import layer_utils
+
+from keras_cv.utils import fill_utils
+
+
+class RandomCutout(layers.Layer):
+    """Randomly cut out rectangles from images and fill them.
+
+    Args:
+        height_factor: a positive float represented as fraction of image height,
+            or as absolute height when factor is an integer. When factor is a tuple
+            of size 2 it represents lower and upper bound for height. For instance,
+            `height_factor=(0.2, 0.3)` results in a height randomly picked in the
+            range `[20% of image height, 30% of image height]`, and
+            `height_factor=(32, 64)` results in a height picked in the range
+            [32, 64]. `height_factor=0.2` results in a height of 20% of image
+            height, and `height_factor=32` results in a height of 32.
+        width_factor: a positive float represented as fraction of image width,
+            or as absolute width when factor is an integer. When factor is a tuple
+            of size 2 it represents lower and upper bound for width. For instance,
+            `width_factor=(0.2, 0.3)` results in a width randomly picked in the
+            range `[20% of image width, 30% of image width]`, and
+            `width_factor=(32, 64)` results in a width picked in the range
+            [32, 64]. `width_factor=0.2` results in a width of 20% of image
+            width, and `width_factor=32` results in a width of 32.
+        fill_mode: Pixels inside the patches are filled according to the given
+            mode (one of `{"constant", "gaussian_noise"}`).
+            - *constant*: Pixels are filled with the same constant value.
+            - *gaussian_noise*: Pixels are filled with random gaussian noise.
+        fill_value: a float represents the value to be filled inside the patches
+            when `fill_mode="constant"`.
+        rate: Float between 0 and 1. The probability of augmenting an input.
+            Defaults to 1.0.
+
+    Sample usage:
+    ```python
+    (images, labels), _ = tf.keras.datasets.cifar10.load_data()
+    random_cutout = keras_cv.layers.preprocessing.RandomCutout(0.5, 0.5, rate=1.0)
+    augmented_images = random_cutout(images)
+    ```
+    """
+
+    def __init__(
+        self,
+        height_factor,
+        width_factor,
+        fill_mode="constant",
+        fill_value=0.0,
+        rate=1.0,
+        seed=None,
+        **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        layer_utils.validate_string_arg(
+            fill_mode,
+            allowable_strings=["constant", "gaussian_noise"],
+            layer_name="RandomCutout",
+            arg_name="fill_mode",
+            allow_none=False,
+            allow_callables=False,
+        )
+
+        if isinstance(height_factor, (tuple, list)):
+            self.height_lower = height_factor[0]
+            self.height_upper = height_factor[1]
+        else:
+            self.height_lower = height_factor
```

Comment:
I'd like to actually default this to 0

Same with width.  I can push this change myself though.